src/pages-and-resources/app-settings-modal/AppSettingsModal.jsx

```diff
@@ -59,8 +61,18 @@ function AppSettingsModal({
 }) {
   const { courseId } = useContext(PagesAndResourcesContext);
   const loadingStatus = useSelector(getLoadingStatus);
+  const appliedSettingsStatus = useSelector(getSavingStatus);
   const appInfo = useModel('courseApps', appId);
   const dispatch = useDispatch();
+  const submitButtonState = appliedSettingsStatus === RequestStatus.IN_PROGRESS ? 'pending' : 'default';
+
+  useEffect(() => {
+    if (appliedSettingsStatus === RequestStatus.SUCCESSFUL) {
+      dispatch(updateSavingStatus({ status: '' }));
+      onClose();
```

Comment:
not sure why you are calling `onClose` here, maybe I need to look a little closer.

Reply:
I'm closing calculator setting modal after successfully saving the settings.